cdap-app-fabric/src/test/java/co/cask/cdap/metadata/MetadataHttpHandlerTest.java

```diff
@@ -119,17 +121,39 @@ public void testProperties() throws Exception {
     properties = getProperties(artifactId);
     Assert.assertEquals(artifactProperties, properties);
 
-    // test search for stream
-    Set<MetadataSearchResultRecord> searchProperties = searchMetadata(Id.Namespace.DEFAULT,
-                                                                      "stKey:stValue", MetadataSearchTargetType.STREAM);
+    // test search for application
     Set<MetadataSearchResultRecord> expected = ImmutableSet.of(
+      new MetadataSearchResultRecord(application)
+    );
+    Set<MetadataSearchResultRecord> searchProperties = searchMetadata(Id.Namespace.DEFAULT,
+                                                                      "aKey:aValue",
+                                                                      MetadataSearchTargetType.APP);
+    Assert.assertEquals(expected, searchProperties);
+    searchProperties = searchMetadata(Id.Namespace.DEFAULT, "multiword:wow1", MetadataSearchTargetType.APP);
+    Assert.assertEquals(expected, searchProperties);
+    searchProperties = searchMetadata(Id.Namespace.DEFAULT, "multiword:woW5", MetadataSearchTargetType.APP);
```

Comment:
also add a verification for the query 'wow5' since that would verify both case-insensitivity and word search?